ratarmount.py

```diff
@@ -884,17 +897,17 @@ def _createIndex(
 
             # fmt: off
             fileInfo = (
-                ""                 ,  # 0 path
-                fname              ,  # 1
-                None               ,  # 2 header offset
-                0                  ,  # 3 data offset
-                fileSize           ,  # 4
-                tarInfo.st_mtime   ,  # 5
-                tarInfo.st_mode    ,  # 6
-                None               ,  # 7 TAR file type. Currently unused but overlaps with mode anyways
-                None               ,  # 8 linkname
-                tarInfo.st_uid     ,  # 9
-                tarInfo.st_gid     ,  # 10
+                ""                                          ,  # 0 path
+                fname                                       ,  # 1
+                None                                        ,  # 2 header offset
+                0                                           ,  # 3 data offset
+                fileSize                                    ,  # 4
+                tarInfo.st_mtime if tarInfo else None       ,  # 5
+                tarInfo.st_mode if tarInfo else None        ,  # 6
+                None                                        ,  # 7 TAR file type. Currently unused but overlaps with mode anyways
+                None                                        ,  # 8 linkname
+                tarInfo.st_uid if tarInfo else None         ,  # 9
```

Comment:
I'm a bit surprised that this works. I would have thought that the SQL scheme checks against None values. I would have filled it with some dummy values like 0 or so, especially for the mode, which should return read permissions at least. Then again, the permissions shouldn't matter as much as they did in FUSE. In the first place, this code location should only be relevant for non-TAR gz files, which theoretically could be accessed directly using `indexed_gzip`. Did you test such files and that is why your are fixing this part or am I misunderstanding something here?

Reply:
I didn't test non-TAR gz files. I'm not clear what you mean by "this code location should only be relevant for non-TAR gz files", because I thought ratarmount is designed to work only with archives (.tar.gz or .tar, etc.) and doesn't work with non-TAR gz files.

I just changed this because tarInfo is not set when the fileobj doesn't have a fileno, so I just wanted to set sensible defaults so the code didn't crash. Your suggestion for dummy values other than None makes sense, though; will change the code to have that!

Reply:
Ah, I see that ratarmount also supports non-TAR gz files (https://github.com/mxmlnkn/ratarmount#compressed-non-tar-files). This is actually a very useful use case for me (as it appears I can use SQLiteIndexedTar to read those files as well)!